packages/core/src/render3/instructions.ts

```diff
@@ -840,6 +842,11 @@ export function directive<T>(index: number, directive?: T, directiveDef?: Direct
     ngDevMode && assertDataInRange(index - 1);
     Object.defineProperty(
         directive, NG_HOST_SYMBOL, {enumerable: false, value: previousOrParentNode});
+
+    if (localName != null) {
+      Object.defineProperty(directive, NG_LOCALNAME_SYMBOL, {enumerable: false, value: localName});
```

Comment:
this is a slow operation and one which is megamorphic. We need to solve this in a way which does not require monkey patching.